api/HTMLButtonElement.json

```diff
@@ -942,10 +942,10 @@
               "version_added": "≤12.1"
             },
             "safari": {
-              "version_added": "3.2"
+              "version_added": "4"
```

Comment:
How confident are you that this wasn't supported in Safari 3? Which platform was Safari 3 tested on?

https://trac.webkit.org/changeset/35381/webkit added the `willValidate` attribute to a bunch of interfaces, and the timing of it is consistent with is being in Safari 3.2 and Safari for iOS 3.

Reply:
I tested these in Mac OS X on Safari 3.2.3 (which is the latest 3.2 release).

Reply:
Continued in https://github.com/mdn/browser-compat-data/pull/.